packages-next/keystone/src/session/index.ts

```diff
@@ -104,15 +104,11 @@ export function withItemData<T extends { listKey: string; itemId: string }>(
             where: { id: session.itemId },
             resolveFields: fieldSelections[session.listKey] || 'id',
           });
-          // If there is no matching item found, return the session without a `data value
-          if (!item) {
-            return session;
-          }
```

Comment:
Note this case never happened because findOne throws when it doesn't find an item